go/vt/tabletserver/endtoend/framework/server.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package framework
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/youtube/vitess/go/sqldb"
+	"github.com/youtube/vitess/go/vt/dbconfigs"
+	"github.com/youtube/vitess/go/vt/mysqlctl"
+	"github.com/youtube/vitess/go/vt/proto/query"
+	"github.com/youtube/vitess/go/vt/proto/topodata"
+	"github.com/youtube/vitess/go/vt/tabletserver"
+	"github.com/youtube/vitess/go/vt/vttest"
+)
+
+var (
+	// BaseConfig is the base config of the default server.
+	BaseConfig tabletserver.Config
+	// Target is the target info for the default server.
+	Target query.Target
+	// DefaultServer is the default server.
+	DefaultServer *tabletserver.TabletServer
+	// ServerAddress is the http URL for the default server.
+	ServerAddress string
+)
+
+// StartDefaultServer starts the default server and initializes
+// all the global variables. This function should only be called
+// once at the beginning of the test.
+func StartDefaultServer(connParams sqldb.ConnParams) error {
+	dbcfgs := dbconfigs.DBConfigs{
+		App: dbconfigs.DBConfig{
+			ConnParams:        connParams,
+			Keyspace:          "vttest",
+			Shard:             "0",
+			EnableRowcache:    true,
+			EnableInvalidator: false,
+		},
+	}
+
+	mysqld := mysqlctl.NewMysqld(
+		"Dba",
+		"App",
+		&mysqlctl.Mycnf{},
+		&dbcfgs.Dba,
+		&dbcfgs.App.ConnParams,
+		&dbcfgs.Repl)
+
+	BaseConfig = tabletserver.DefaultQsConfig
+	BaseConfig.RowCache.Binary = vttest.MemcachedPath()
+	BaseConfig.RowCache.Socket = path.Join(os.TempDir(), "memcache.sock")
+	BaseConfig.EnableAutoCommit = true
+
+	Target = query.Target{
+		Keyspace:   "vttest",
+		Shard:      "0",
+		TabletType: topodata.TabletType_MASTER,
+	}
+
+	DefaultServer = tabletserver.NewTabletServer(BaseConfig)
+	DefaultServer.Register()
+	err := DefaultServer.StartService(&Target, &dbcfgs, nil, mysqld)
+	if err != nil {
+		return fmt.Errorf("could not start service: %v\n", err)
+	}
+
+	// Start http service.
+	ln, err := net.Listen("tcp", ":0")
```

Comment:
I thought about that route. But in the past, we've had bugs where the json output was corrupted due incorrect rendering. Since this is an end to end test, I thought we should make sure that part works also.